routing/route/route.go

```diff
@@ -267,6 +271,11 @@ func NewRouteFromHops(amtToSend lnwire.MilliSatoshi, timeLock uint32,
 func (r *Route) ToSphinxPath() (*sphinx.PaymentPath, error) {
 	var path sphinx.PaymentPath
 
+	// Check maximum route length.
+	if len(r.Hops) > sphinx.NumMaxHops {
```

Comment:
Could we move the `len(rt.Hops) == 0` check from `generateSphinxPacket` to here as well? It returns an error from the `route` package anyway.

Reply:
Nice suggestion, done